### DISCUSSION TITLE
Whether should we force ENSG ID to be unique identifier.

### DISCUSSION THREAD

**Discussion by hsun3163** [StatFunGen/xqtl-protocol#135] [General]
After the current annotate_coord step, no records remains due to the log2cpm have gene name as the first column, instead of gene ENSG ID.

This issue can easily be fixed by changing an option of gtf_to_tss_bed function.
`bed_template_df = qtl.io.gtf_to_tss_bed(${_input[1]:ar}, feature='transcript',phenotype_id = ${phenotype_id_type} )`

However, the output bed of such will retained the gene names, instead of keeping the ENSG ID. This leads to the question in the title, shall we find a way to force it all to be ENSG ID?

I leaned toward of using ensg ID as the unique identifier to avoid ambiguity.

**Comment by marcora** [comment]
I often use a ENSGENEID:GENESYM "merge" to avoid ambiguity yet have some
level of legibility.
